kwasm-lemmas.md

```diff
@@ -131,6 +141,14 @@ x * m + y mod n = x * (k * n) + y mod n = y mod n
        andBool N >Int 0
        andBool M modInt N ==Int 0
       [simplification]
+
+    rule #wrap(N, #wrap(M, X) +Int Y) => #wrap(N, X +Int Y)
+      requires N <=Int M
+      [simplification]
+
+    rule #wrap(N, X +Int #wrap(M, Y)) => #wrap(N, X +Int Y)
+      requires N <=Int M
+      [simplification]
```

Comment:
For instance, you could have one lemma which says `rule #wrap(M, Y) +Int X => X +Int #wrap(M, Y) [symbolic(M, Y, X)]` and another `rule (X <<Int N) +Int Y => Y +Int (X <<Int N) [symbolic(X, N, Y)]`, and then all your other lemmas could be written instead with just one of the commutative options (instead of having to have both).

Reply:
Issue #315 addresses this. There are several other lemmas like this, so I'd rather hit them all in one refactor rather rhan making it part of this PR.